### PR TITLE
feat: add support for e-Money Gas Prices

### LINKED ISSUE #202: [e-Money] Get minimum gas prices
## Feature description 

Log the minimum gas prices from x/authority

```
emd q authority gas-prices
```

```
{
  "min_gas_prices": [
    {
      "denom": "echf",
      "amount": "0.530000000000000000"
    },
    {
      "denom": "edkk",
      "amount": "3.700000000000000000"
    },
    {
      "denom": "eeur",
      "amount": "0.500000000000000000"
    },
    {
      "denom": "enok",
      "amount": "6.100000000000000000"
    },
    {
      "denom": "esek",
      "amount": "5.500000000000000000"
    },
    {
      "denom": "ungm",
      "amount": "1.000000000000000000"
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by huichiaotsou**
## Description
Fixes #202 

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit tests.  
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by MonikaCat** [comment]
Tested on my local ACK

**Comment by huichiaotsou** [comment]
Hello @RiccardoM , I have completed the e-Money min gas prices module with hasura metadata and unit tests, but I'm not sure why there's a linter error :-/

**Review by rmontagnin — APPROVED** [review]
utACK

**Comment by rmontagnin** [comment]
Can I get a new confirmation from @MonikaCat that this works please?

**Comment by MonikaCat** [comment]
> Can I get a new confirmation from @MonikaCat that this works please?

I tested it on my local, it works. However, should we add periodic operation to query the gas prices once a day or so?

**Comment by rmontagnin** [comment]
> I tested it on my local, it works. However, should we add periodic operation to query the gas prices once a day or so?

To be honest, I have no idea how this module works and what it represents. I see that @huichiaotsou added a message handler for gas prices. Is that necessary, or should it be replace with a periodic query instead?

**Comment by huichiaotsou** [comment]
> > I tested it on my local, it works. However, should we add periodic operation to query the gas prices once a day or so?
> 
> To be honest, I have no idea how this module works and what it represents. I see that @huichiaotsou added a message handler for gas prices. Is that necessary, or should it be replace with a periodic query instead?

So e-Money has an authority module in which they set the gas prices : https://github.com/e-money/em-ledger/blob/master/x/authority/handler.go#L31

which is why I used message handler to fetch the msg SetGasPrices. I didn't set up periodic ops as I thought it wouldn't be updated so frequently ? 😆

**Comment by MonikaCat** [comment]
Yes this approach will work if we are setting up bdjuno from genesis time. Bdjuno will process the genesis file and save the gas values in db. However, if we update bdjuno that is currently running with it, we will need to rely on/wait for the gas prices to be updated before we can store the values in db. Do you know how often they are updated? 
Also, I can't seem to find any transactions with this msg in hasura ```/em.authority.v1.MsgSetGasPrices```? Is this the correct msg type?

File: modules/authority/handle_periodic_operations.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	if _, err := scheduler.Every(1).Day().At("00:00").StartImmediately().Do(func() {
+		utils.WatchMethod(func() error { return updateMinGasPrices(authorityclient, db) })
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// updateInflation fetches from the REST APIs the latest value for the
```
Comment (MonikaCat) [inline modules/authority/handle_periodic_operations.go]: I guess this should be ```updateMinGasPrices```

Comment (huichiaotsou) [reply modules/authority/handle_periodic_operations.go]: thanks~ it's updated :D

**Comment by MonikaCat** [comment]
Tested on my local with periodic ops added, ACK

File: modules/authority/module.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	return HandleGenesis(genesisDoc, appState, m.encodingConfig.Marshaler, m.db)
+}
+
+// RegisterPeriodicOperations implements modules.PeriodicOperationsModule
+func (m *Module) RegisterPeriodicOperations(scheduler *gocron.Scheduler) error {
+	return RegisterPeriodicOps(scheduler, m.authorityClient, m.db)
+}
+
+// HandleMsg implements modules.MessageModule
+func (m *Module) HandleMsg(_ int, msg sdk.Msg, tx *types.Tx) error {
```
Comment (rmontagnin) [inline modules/authority/module.go]: Should we delete this since we are already updating them periodically?

Comment (huichiaotsou) [reply modules/authority/module.go]: hello @RiccardoM I have removed the handle msg :D

**Review by rmontagnin — APPROVED** [review]

**Pull request merged**